### PR TITLE
feat: add prosemd-lsp for markdown

### PR DESCRIPTION
**Pull request by dsully**
Add prosemd-lsp, a LSP for Markdown.

### CODE REVIEW

File: lua/lspconfig/server_configurations/prosemd_lsp.lua [hunk 1]
```diff
@@ -0,0 +1,22 @@
+local util = require 'lspconfig.util'
+
+return {
+  default_config = {
+    cmd = { 'prosemd-lsp', '--stdio' },
+    filetypes = { 'markdown' },
+    root_dir = function(fname)
+      return util.find_git_ancestor(fname) or vim.fn.getcwd()
```
Comment (mjlbach) [inline lua/lspconfig/server_configurations/prosemd_lsp.lua:8]: `root_dir = util.find_git_ancestor`

File: lua/lspconfig/server_configurations/prosemd_lsp.lua [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    end,
+    single_file_support = true,
+  },
+  docs = {
+    description = [[
+https://github.com/kitten/prosemd-lsp
```
Comment (mjlbach) [inline lua/lspconfig/server_configurations/prosemd_lsp.lua:14]: Installation instructions?

File: lua/lspconfig/server_configurations/prosemd_lsp.lua [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+
+An experimental LSP for Markdown.
+]],
+    default_config = {
+      root_dir = [[find_git_ancestor(fname) or vim.fn.getcwd()]],
```
Comment (mjlbach) [inline lua/lspconfig/server_configurations/prosemd_lsp.lua:19]: Update when you address my other comment.

**Comment by dsully** [comment]
Since prosemd-lsp has manual install instructions, I pointed to those. Is that sufficient?

**Comment by mjlbach** [comment]
Yup

**Pull request merged**